charms/admission-webhook/src/charm.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+import os
+from base64 import b64encode
+from pathlib import Path
+from subprocess import run
+
+import yaml
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, MaintenanceStatus
+
+from charmhelpers.core import hookenv
+from oci_image import OCIImageResource, OCIImageResourceError
+
+
+class AdmissionWebhookCharm(CharmBase):
+    def __init__(self, framework, key):
+        super().__init__(framework, key)
+        self.image = OCIImageResource(self, 'oci-image')
+        self.framework.observe(self.on.install, self.set_pod_spec)
+        self.framework.observe(self.on.upgrade_charm, self.set_pod_spec)
+
+    def set_pod_spec(self, event):
+        if not self.model.unit.is_leader():
+            print('Not a leader, skipping set_pod_spec')
```

Comment:
Better to use logging, because you can get to use different levels according to what you want to inform...

Just do...

```
logger = logging.getLogger(__name__)
```

at module level before the class definition, and then use it like

```
logger.info('Not a leader, skipping set_pod_spec')
```

Note that if INFO, the user will be able to see it in the logs WITHOUT doing anything extra,..